include/swift/AST/DiagnosticsParse.def

```diff
@@ -143,6 +143,9 @@ WARNING(lex_editor_placeholder_in_playground,none,
 // Declaration parsing diagnostics
 //------------------------------------------------------------------------------
 
+NOTE(note_in_decl_extension,none,
+     "in %select{declaration|extension}0 of %1:", (bool, Identifier))
```

Comment:
Nitpick: please drop the colon, since it's not coming before the warning anymore.